chef/automate
# [infra-proxy-service] GET - API changes for the policyfile details solution dependencies attr

### :nut_and_bolt: Description: What code changed, and why?
we need to show the solution dependencies for the policy file details as per the policy file name and version.
### :chains: Related Resources
https://github.com/chef/automate/issues/5410
### :+1: Definition of Done
I have added changes to show the solution dependencies with the policy file name and version, now we can see the all dependent cookbook name and versions which are dependent.
### :athletic_shoe: How to Build and Test the Change

### :white_check_mark: Checklist
Steps to build:
```
rebuild components/infra-proxy-service
rebuild components/automate-gateway
```
if you don't have sample data then you need to add first
1.  Ensure Chef Infra Server is running. If it is not start it with `start_chef_server`
2. First you need to run the `infra_service_load_chef_repo`
3. Then run `converge_policyfile_chef_client`

Steps to test:

Add the chef-server test environment credentials and extract the SERVER and ORG IDs.
Policy group details API - 
request 
```
curl -sSkH "api-token: $(get_admin_token)"  https://a2-dev.test/api/v0/infra/servers/SERVER/orgs/ORG/policyfiles/NAME?revision_id=YOUR_REVISION_ID
```
response 
```
{
    "name":"vinay-dev",
    "policy_group":"",
    "revision_id":"34148f800c4fc00001b7d3db03c03cf91e6f5d2dabfee56bc15028e21e8d9dc5",
    "run_list":["recipe[pantry::default]","recipe[mycookbook::default]"],
    "named_run_list":...data,
    "included_policy_locks":...data,
    "cookbook_locks":...data,
    "default_attributes":"{\"chocolatey\":{\"packages\":[]},\"homebrew\":{\"taps\":[]}}",
    "override_attributes":"{}","expanded_run_list":[],
    "solution_dependecies":[{"name":"build-essential","version":"= 8.2.1","dependencies":[{"name":"seven_zip","version":"\u003e= 0.0.0"}]
}
```

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
